packages/protocol/contracts/bridge/Bridge.sol

```diff
@@ -352,6 +352,18 @@ contract Bridge is EssentialContract, IBridge {
             revert B_NON_RETRIABLE();
         }
 
+        // We check _message.gasLimit > gasleft() to make sure we not only need to bridge invocation
+        // call to succeeed, we also need it to succeed with a gas limit no smaller than the
+        // message's gasLimit.
+        // Also if caller is not the destOwner, we need to be sure there is enough gas for rest of
+        // the code to finish execution.
+        if (
+            (_message.gasLimit != 0 && _message.gasLimit > gasleft())
```

Comment:
if gasLimit == 0, msg.sender must be the destOwner, so the checks here an be simplified:

Reply:
```
if (_message.gasLimit != 0 &&
msg.sender != _message.destOwner &&
_message.gasLimit > (gasleft() * 63) >> 6) {
```